src/dotnet-install.ps1

```diff
@@ -289,16 +289,34 @@ function GetHTTPResponse([Uri] $Uri)
             $Task = $HttpClient.GetAsync("${Uri}${FeedCredential}").ConfigureAwait("false");
             $Response = $Task.GetAwaiter().GetResult();
             if (($Response -eq $null) -or (-not ($Response.IsSuccessStatusCode))) {
-                 # The feed credential is potentially sensitive info. Do not log FeedCredential to console output.
-                $ErrorMsg = "Failed to download $Uri."
+                # The feed credential is potentially sensitive info. Do not log FeedCredential to console output.
+                $ErrorMsg = ""
                 if ($Response -ne $null) {
-                    $ErrorMsg += "  $Response"
+                    $ErrorMsg += "$Response"
                 }
-
                 throw $ErrorMsg
             }
 
             return $Response
+
+        }
+        catch {
```

Comment:
You might want to catch HTTP Request Exception here, not to rely on status code. Then Status Code will be returned as one of the properties.
It is possible to catch certain types of Exception in PowerShell: https://docs.microsoft.com/powershell/scripting/learn/deep-dives/everything-about-exceptions?view=powershell-7.1#catching-typed-exceptions

Reply:
Nice idea, I suppose I'd like to specialize the catch here. But I am not sure it will immediately help to remove parsing.
First, Property Status Code for HttpRequestException is a 5.0 feature, not sure I can use it:
https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httprequestexception.statuscode?view=net-5.0#System_Net_Http_HttpRequestException_StatusCode
And secondly, we need also to have this property for the exception we throw explicitly like in row 298: `throw $ErrorMsg`. So, we either need to write a custom exception class and throw it or to find the existing one to create there.

Reply:
Sorry for confusion. Previous message intend to say `throw` instead of catch. You might want to use: https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpresponsemessage.ensuresuccessstatuscode?view=net-5.0 method to do it.
Having StatusCode property is unfortunate, probably then it is easier to keep it as is as anyway parsing is required. 

Is it an option to save the http code to variable to avoid parsing? (similar to bash script). In this method the status code is still available via property and putting it to string, and parsing again doesn't look as a good idea.  Should we consider updating the method just to return the response and do error handling outside?

Reply:
Fixed. Used the `Data` property of the exception to preserve the status code, as discussed.